fish-pepper/run-java-sh/fp-files/run-java.sh

```diff
@@ -81,6 +106,13 @@ load_env() {
     source "${script_dir}/${run_env_sh}"
   fi
 
+  if [ -z "${MAX_CORES}" ]; then
```

Comment:
Maybe we can choose a bit more specific env var name. Currently most of them are prefixed with "JAVA_" but in this casae its not sth a user can set, its more a read-only variable. But maybe `JAVA_MAX_CORES_AVAILABLE` still fits ?